src/app/public/modules/modal-provider/types/modal-provider.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  EventEmitter
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs';
+
+import {
+  SkyModalProviderCloseArgs
+} from './modal-provider-close-args';
+
+export interface SkyModalProvider {
+
+  type: string;
+  closeCallback: Observable<SkyModalProviderCloseArgs>;
+  events?: { [key: string]: EventEmitter<any> };
+  open(paramObject?: any): any;
```

Comment:
What does the `open` property do?

Reply:
That is the function that the consuming SPA/Library calls to open the modal. It is implemented by the provider itself. 

It has an `any` param because this interface doesn't know what parameters each modal might need and it has an `any` return type because we don't know what each modal might want to return.